internal/characters/neuvillette/charge.go

```diff
@@ -0,0 +1,289 @@
+
+	endLag = frames.InitAbilSlice(51)
+	endLag[action.ActionWalk] = 36
+	endLag[action.ActionCharge] = 30
+	endLag[action.ActionSwap] = 27
+	endLag[action.ActionBurst] = 0
+	endLag[action.ActionSkill] = 0
+	endLag[action.ActionDash] = 0
+	endLag[action.ActionJump] = 0
+
+	earlyCancelEndLag = frames.InitAbilSlice(5000)
+	earlyCancelEndLag[action.ActionBurst] = 0
+	earlyCancelEndLag[action.ActionSkill] = 0
+	earlyCancelEndLag[action.ActionDash] = 0
+	earlyCancelEndLag[action.ActionJump] = 0
+}
+
+func (c *char) legalEvalFindDroplets() int {
+	droplets := c.getSourcewaterDroplets()
+
+	// TODO: If droplets time out before the "droplet check" it doesn't count.
+	indices := c.Core.Combat.Rand.Perm(len(droplets))
+	orbs := 0
+	for _, ind := range indices {
+		g := droplets[ind]
+		c.consumeDroplet(g)
+		orbs += 1
+		if orbs >= 3 {
+			break
+		}
+	}
+	c.Core.Combat.Log.NewEvent(fmt.Sprint("Picked up ", orbs, " droplets"), glog.LogCharacterEvent, c.Index)
+	return orbs
+}
+
+func (c *char) ChargeAttack(p map[string]int) (action.Info, error) {
+	if c.chargeEarlyCancelled {
+		return action.Info{}, fmt.Errorf("%v: Cannot early cancel Charged Attack: Equitable Judgement with Charged Attack", c.CharWrapper.Base.Key)
+	}
+	// there is a windup out of dash/jump/walk/swap. Otherwise it is rolled into the Q/E/CA/NA -> CA frames
+	windup := 0
+	switch c.Core.Player.CurrentState() {
+	case action.Idle, action.DashState, action.JumpState, action.WalkState, action.SwapState:
+		windup = 14
+	}
+
+	if p["short"] != 0 {
+		return c.chargeAttackShort(windup)
+	}
+
+	return c.chargeAttackJudgement(p, windup)
+}
+
+func (c *char) chargeAttackJudgement(p map[string]int, windup int) (action.Info, error) {
+	c.chargeJudgeDur = 0
+	c.nextTickTime = getChargeJudgementHitmarkDelay(0)
+	// current framework doesn't really support actions getting shorter, so the legal eval is set to 0, but it may increase later
+	chargeLegalEvalLeft := 0
+
+	c.QueueCharTask(func() {
+		chargeLegalEvalLeft = initialLegalEvalDur
+		orbs := c.legalEvalFindDroplets()
+		chargeLegalEvalLeft -= dropletLegalEvalReduction[orbs]
+
+		c.chargeAi = combat.AttackInfo{
+			ActorIndex: c.Index,
+			Abil:       chargeJudgementName,
+			AttackTag:  attacks.AttackTagExtra,
+			ICDTag:     attacks.ICDTagExtraAttack,
+			ICDGroup:   attacks.ICDGroupDefault,
+			StrikeType: attacks.StrikeTypeDefault,
+			Element:    attributes.Hydro,
+			Durability: 25,
+			FlatDmg:    chargeJudgement[c.TalentLvlAttack()] * c.MaxHP(),
+		}
+
+		c.chargeJudgeStartF = c.Core.F + chargeLegalEvalLeft
+		c.chargeJudgeDur = 173
+
+		if c.Base.Cons >= 6 {
+			c.QueueCharTask(c.c6DropletCheck(c.chargeJudgeStartF), chargeLegalEvalLeft)
+			c.QueueCharTask(c.c6(c.chargeJudgeStartF), chargeLegalEvalLeft)
+		}
+
+		ticks, ok := p["ticks"]
+		if !ok {
+			ticks = -1
+		} else if ticks < 0 {
+			ticks = 0
+		}
+
+		c.QueueCharTask(c.chargeJudgementTick(c.chargeJudgeStartF, 0, ticks, false), chargeLegalEvalLeft+getChargeJudgementHitmarkDelay(0))
```

Comment:
change this to:
```
// cannot use hitlag affected queue because the logic just does not work then
// -> can't account for possible hitlag delaying the update of the anim length (sim moves on to next action, but ticks continue)
// start counting at 1 for correct number of ticks when supplying ticks param
c.Core.Tasks.Add(c.chargeJudgementTick(c.chargeJudgeStartF, 1, ticks, false), chargeLegalEvalLeft+getChargeJudgementHitmarkDelay(0))
```